gradle/lib/dependabot/gradle/file_updater/lockfile_updater.rb

```diff
@@ -0,0 +1,110 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "shellwords"
+
+require "dependabot/gradle/file_parser"
+require "dependabot/gradle/file_updater"
+
+module Dependabot
+  module Gradle
+    class FileUpdater
+      class LockfileUpdater
+        extend T::Sig
+
+        sig { params(dependency_files: T::Array[Dependabot::DependencyFile]).void }
+        def initialize(dependency_files:)
+          @dependency_files = dependency_files
+        end
+
+        sig { params(build_file: Dependabot::DependencyFile).returns(T::Array[Dependabot::DependencyFile]) }
+        def update_lockfiles(build_file)
+          local_lockfiles = dependency_files.select do |file|
+            file.directory == build_file.directory && file.name.end_with?(".lockfile")
+          end
+
+          # If we don't have any lockfiles in the build files don't generate one
+          return dependency_files unless local_lockfiles.any?
+
+          updated_files = dependency_files.dup
+          SharedHelpers.in_a_temporary_directory do |temp_dir|
+            populate_temp_directory(temp_dir)
+            cwd = File.join(temp_dir, build_file.directory, build_file.name)
+            cwd = File.dirname(cwd)
+
+            # Create gradle.properties file with proxy settings
+            # Would prefer to use command line arguments, but they don't work.
+            properties_filename = File.join(temp_dir, build_file.directory, "gradle.properties")
+            write_properties_file(properties_filename)
+
+            command_parts = [
+              "gradle",
+              "dependencies",
+              "--no-daemon",
+              "--write-locks"
+            ]
+            command = Shellwords.join(command_parts)
+
+            Dir.chdir(cwd) do
+              SharedHelpers.run_shell_command(command, cwd: cwd)
```

Comment:
Are there any interesting error messages that might come from this?  Potentially a network call that might return 401 or 403?  If so, those should be captured/parsed and turned into `private_source_authentication_error`.

Reply:
Unfortunately, the error messages out of the Gradle CLI were pretty opaque. The only way I found to get better detail was to define a custom task which manually ran the dependencies check per config, but that requires editing the build.gradle file which turns into a whole mess.

We also explicitly don't want to fail if the gradle cli fails since there's lots of reasons that could happen and we still want to update people's build.gradle in that case.